libs/frc20_abi/src/lib.sw

```diff
@@ -0,0 +1,22 @@
+library frc20_abi;
+
+use std::u256::U256;
+
+abi Token {
+    #[storage(read)]
+    /// Get the total supply of the token.
+    /// MUST track all mint and burn operations.
+    /// MAY store the amount as another type (i.e: 'u64') internally.
```

Comment:
```suggestion
    /// Get the total supply of the token.
    /// MUST track all mint and burn operations.
    /// MAY store the amount as another type (i.e: 'u64') internally.
    #[storage(read)]
```

Unsure about the last 2 comments.

What is the purpose of telling the reader that it must track both? Under which conditions would that not be the case? If anything this should only be a getter, and this has the `read` but not `write` annotation, so this doesn't really track anything except return the value. It does indicate that both operations must be accounted for; however, I do not see why that needs to be explicitly stated.

The second comment regarding the type and the actual return type. This should really be set here rather than suggesting that another type may be used. If this is meant to be a standard then allowing users to mix and match types does not conform to the idea of a standard.
I'd probably just go with a u64 for simplicity.

(So I'd remove the last 2 comments and change the return type)

Reply:
Thanks @Braqzen. Yeah, I'm removing the last 2 comments; while it is normal to specify this sort of thing in a token standard, they don't belong on a simple getter.
See [erc20](https://eips.ethereum.org/EIPS/eip-20#specification) for an example, ie `transfer` which specifies which event must be emitted.
I'd consider removing the comment about types. 
We discussed this at length on Slack (which is not to say there was 100% consensus), and a U256 allows for tokens with  extremely large supplies, and/or very high precision needs. The standard specifies the function signatures, but how the user implements storage details is up to them.